src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,236 @@
+/**
+ * @file methods/kernel_svm/kernel_svm_impl.hpp
+ * @author Himanshu Pathak
+ *
+ * Implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "kernel_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    const double delta,
+    const double C,
+    const bool fitIntercept,
+    const size_t max_iter,
+    const double tol) :
+    numClasses(numClasses),
+    delta(delta),
+    C(C),
+    fitIntercept(fitIntercept),
+    max_iter(max_iter)
+{
+  Train(data, labels, numClasses, max_iter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const size_t inputSize,
+    const size_t numClasses,
+    const double delta,
+    const double C,
+    const bool fitIntercept) :
+    inputSize(inputSize),
+    numClasses(numClasses),
+    delta(delta),
+    C(C),
+    fitIntercept(fitIntercept)
+{
+  // No training to do here.
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVM<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    const size_t max_iter,
+    const double tol)
+{
+  arma::Row<int> label(data.n_cols);
+  for (size_t i = 0; i< data.n_cols; i++)
+  {
+    if(labels(i) == 0)
+      label(i) = -1;
```

Comment:
Hey @saksham189 , @zoq I checked some implementation everywhere other label is used as `-1`. So, I tried it locally I got the satisfactory result. So, if I get all the builds green with this method I will move forward with this implementation also it is only specific when using `LinearKernel` with `smo` :grin: not for other kernel.

Reply:
can you explain why this is required. Do you think the user should be providing the labels as 0 1 or -1 1 ?

Reply:
Probably we should use `0` and `1` here---the rest of mlpack generally uses `arma::Row<size_t>` to specify labels. :+1:

Reply:
Will change it in next commit. :+1: